rocAL/include/readers/image/numpy_data_reader.h

```diff
@@ -0,0 +1,134 @@
+/*
+Copyright (c) 2024 Advanced Micro Devices, Inc. All rights reserved.
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+#pragma once
+#include <dirent.h>
+
+#include <memory>
+#include <mutex>
+#include <string>
+#include <vector>
+
+#include "pipeline/commons.h"
+#include "pipeline/timing_debug.h"
+#include "readers/image/image_reader.h"
+
+class NumpyDataReader : public Reader {
+   public:
+    //! Looks up the folder which contains the files, amd loads the numpy files
+    /*!
+     \param desc  User provided descriptor containing the files' path.
+    */
+    Reader::Status initialize(ReaderConfig desc) override;
+    //! Reads the next resource item
+    /*!
+     \param buf User's provided buffer to receive the loaded numpy data
```

Comment:
define max_size param

Reply:
Not used in numpy reader so ignoring param